lib/entice/logic/vitals.ex

```diff
@@ -4,61 +4,139 @@ defmodule Entice.Logic.Vitals do
     mana: 50, max_mana: 70)
 
+  defmodule Morale, do: defstruct(
+    morale: 0)
 
   def register(entity_id),
-  do: Entity.put_behaviour(entity_id, Vitals.Behaviour, [])
+  do: Entity.put_behaviour(entity_id, Vitals.AliveBehaviour, [])
+
+
+  def unregister(entity_id) do
+    Entity.remove_behaviour(entity_id, Vitals.AliveBehaviour)
+    Entity.remove_behaviour(entity_id, Vitals.DeadBehaviour)
+  end
 
+  # External API
 
-  def unregister(entity_id),
-  do: Entity.remove_behaviour(entity_id, Vitals.Behaviour)
+  @doc """
+  Used to make damage on a Entity
+  Damage is applied to the Entity
+  Entity can die
+  """
+  def damage(entity, amount),
+  do: Coordination.notify({:vitals_entity_damage, ammount}, entity)
 
+  @doc """
+  Heals a entity with the given amount
+  Health of Entity is recalculated
+  """
+  def heal(entity, amount),
+  do: Coordination.notify({:vitals_entity_heal, ammount}, entity)
 
-  defmodule Behaviour do
+  @doc """
+  Resurrects an entity with a Percentage of Life Points and Energy
+  New Health and Energy is dependent of Morale
+  """
+  def resurrect(entity, percent_health, percent_energy),
+  do: Coordination.notify({:vitals_entity_resurrect, percent_health, percent_energy}, entity)
+
+  defmodule AliveBehaviour do
     use Entice.Entity.Behaviour
     alias Entice.Logic.Vitals.Health
     alias Entice.Logic.Vitals.Energy
+    alias Entice.Logic.Vitals.Morale
     alias Entice.Logic.Player.Level
 
+    def init(%Entity{attributes: %{Level => %Level{level: _}}} = entity, {:entity_resurrected, percent_health, percent_energy}) do
+      {_, max_health: max_health} = get_max_health(entity)
+      resurrected_health = max_health / 100 * percent_health
 
-    def init(entity, _args) do
-      {:ok, entity |> put_attribute(get_max_health(entity))
-                   |> put_attribute(get_max_energy(entity))}
+      {_, max_mana: max_mana} = get_max_energy(entity)
+      resurrected_mana = max_mana / 100 * percent_energy
 
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: resurrected_health, max_health: max_health} end)
+                   |> update_attribute(Energy, fn _ -> %Energy{mana: resurrected_mana, max_mana: max_mana} end)}
     end
 
+    def init(%Entity{attributes: %{Level => %Level{level: _}}} = entity, _args) do
+      {:ok, entity |> put_attribute(get_max_health(entity))
+                   |> put_attribute(get_max_energy(entity))
+                   |> put_attribute(%Morale{morale: 0})}
+    end
+
+    def terminate(:remove_handler, entity) do
+      {:ok, entity}
+    end
 
     def terminate(_reason, entity) do
       {:ok, entity |> remove_attribute(Health)
-                   |> remove_attribute(Energy)}
+                   |> remove_attribute(Energy)
+                   |> remove_attribute(Morale)}
+    end
+
+    def handle_event({:vitals_entity_damage, amount}, %Entity{attributes: %{Health => %Health{health: health, max_health: _}}} = entity) do
```

Comment:
no need to match on `max_health: _` if we don't need its value anyway - matching on maps does not need to include all the possible map keys in order to match